jenkins/Makefile

```diff
@@ -1,13 +1,77 @@
+#  all (the default):  builds the CI image and a Fedora-based development image
+#  image-ci:           builds the CI image
+#  image-fedora:       builds the Fedora-based development image
+#  image-rpmbuild-all: builds all the default containers for building RPMs
+#  image-rpmbuild-rhel-7:  builds a single RPM build container (e.g., for RHEL-7)
+#  push-ci:            pushes the CI image to the registry
+#  push-fedora:        pushes the Fedora-based development image to the registry
+#  push-rpmbuild-all:  pushes all the default RPM build containers to the registry
+#
+
+# Default list of distributions for which to build containers in which to build RPMs
+RPMBUILD_BASEIMAGE_DEFAULTS = \
+	rhel-9 rhel-8 rhel-7 \
+	fedora-36 fedora-35 fedora-34 \
+	centos-9 centos-8
+
+# Registry where the resulting RPM build container images are to be stored
+RPMBUILD_IMAGE_REPO = images.paas.redhat.com/pbench
+
+# Templates for the various distributions' base images.  For the moment, for a
+# given distribution, all of the versions can be found by tweaking the image
+# tag; so, to generate the appropriate repository for the base image, we just
+# have to pick the appropriate template and insert the requested version, which
+# is exactly what happens when RPMBUILD_BASEIMAGE_REPO is expanded.
 #
+# Note that registry.access.redhat.com is moving to registry.redhat.io where it
+# will no longer permit anonymous access, but it works for the moment.
+VER := __version__
+RPMBUILD_BASEIMAGE_rhel = registry.access.redhat.com/ubi${VER}:latest
+RPMBUILD_BASEIMAGE_fedora = quay.io/fedora/fedora:${VER}
+RPMBUILD_BASEIMAGE_centos = quay.io/centos/centos:stream${VER}
+RPMBUILD_BASEIMAGE_REPO = $(subst ${VER},${DIST_VERSION},${RPMBUILD_BASEIMAGE_${DIST_NAME}})
+
+# List of packages required to build RPMs, to be installed in the build
+# containers.  (The jinja2 CLI is not available as an RPM on RHEL, so we
+# install it via pip for all platforms, to keep things simple.)
+RPMBUILD_PACKAGES = git make python3-jinja2 python3-pip rpmlint rpm-build
+
+# Default package manager tool (may be overridden for some distros)
+PKGMGR = dnf
 
 BRANCH := $(shell cat ./branch.name)
 
+# Compose lists of potential distro-version strings with versions 0-99
+# (e.g., `_ALL_centos_VERSIONS` including "centos-7", "centos-8", "centos-9",
+# `_ALL_fedora_VERSIONS` including "fedora-32", "fedora-33", and "fedora-34",
+# and `_DISTROS` concatenating the contents of all the lists) which will be
+# used to drive pattern matching on distro-based target rules.
+_DIGITS := 0 1 2 3 4 5 6 7 8 9
+_NUMBERS := $(patsubst 0%,%,$(foreach tens,${_DIGITS},$(foreach ones,${_DIGITS},${tens}${ones})))
+_ALL_DISTRO_NAMES := centos fedora rhel
+_ADV_TMPL = _ALL_${d}_VERSIONS := $(foreach v,${_NUMBERS},${d}-${v}) # template for setting version lists
+$(foreach d,${_ALL_DISTRO_NAMES},$(eval $(call _ADV_TMPL, ${d})))  # set _ALL_centos_VERSIONS, etc.
+_DISTROS := $(foreach d,${_ALL_DISTRO_NAMES},${_ALL_${d}_VERSIONS})
```

Comment:
And ... this way we get to review this here before it gets moved into your `util.mk`?? And hasn't this existed for a while? Was it somewhere else?